.buildkite/scripts/steps/esql_grammar_sync.sh

```diff
@@ -0,0 +1,131 @@
+
+  # Replace the line containing "lexer grammar" with "lexer grammar esql_lexer;"
+  sed -i -e 's/lexer grammar.*$/lexer grammar esql_lexer;/' "$destination_file" 
+
+  # Insert "options { caseInsensitive = true; }" one line below
+  sed -i -e '/lexer grammar esql_lexer;/a\
+  options { caseInsensitive = true; }' "$destination_file" 
+
+  echo "File copied and modified successfully."
+}
+
+synchronize_parser_grammar () {
+  license_header="$1"
+  source_file="$PARENT_DIR/elasticsearch/x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4"
+  destination_file="./packages/kbn-monaco/src/esql/antlr/esql_parser.g4"
+
+  # Copy the file
+  cp "$source_file" "$destination_file" 
+
+  # Insert the license header
+  temp_file=$(mktemp)
+  printf "%s\n\n// DO NOT MODIFY THIS FILE BY HAND. IT IS MANAGED BY A CI JOB.\n\n%s" "$license_header" "$(cat ${destination_file})" > "$temp_file"
+  mv "$temp_file" "$destination_file"
+
+  # Replace the line containing "parser grammar" with "parser grammar esql_parser;"
+  sed -i -e 's/parser grammar.*$/parser grammar esql_parser;/' "$destination_file" 
+
+  # Replace options {tokenVocab=EsqlBaseLexer;} with options {tokenVocab=esql_lexer;}
+  sed -i -e 's/options {tokenVocab=EsqlBaseLexer;}/options {tokenVocab=esql_lexer;}/' "$destination_file" 
+
+  echo "File copied and modified successfully."
+}
+
+report_main_step () {
+  echo ""
+  echo "-------------------------------------------------"
+  echo "MAIN STEP: $1"
+  echo "-------------------------------------------------"
+}
+
+main () {
+  cd "$PARENT_DIR" 
+
+  report_main_step "Cloning repositories"
+
+  rm -rf elasticsearch
+  git clone https://github.com/elastic/elasticsearch --depth 1 
+
+  rm -rf open-source
+  git clone https://github.com/elastic/open-source --depth 1 
+
+  cd "$KIBANA_DIR" 
+
+  license_header=$(cat "$PARENT_DIR/open-source/legal/elastic-license-2.0-header.txt")
+
+  report_main_step "Synchronizing lexer grammar..."
+  synchronize_lexer_grammar "$license_header"
+
+  report_main_step "Synchronizing parser grammar..."
+  synchronize_parser_grammar "$license_header"
+
+  # Check for differences
+  set +e
+  git diff --exit-code --quiet "$destination_file"
+  set -e
+
+  if [ $? -eq 0 ]; then
+    echo "No differences found. Our work is done here."
+    exit
+  fi
+
+  report_main_step "Differences found. Checking for an existing pull request."
+
+  KIBANA_MACHINE_USERNAME="kibanamachine"
+  git config --global user.name "$KIBANA_MACHINE_USERNAME"
+  git config --global user.email '42973632+kibanamachine@users.noreply.github.com'
+
+  PR_TITLE='[ES|QL] Update grammars'
+  PR_BODY='This PR updates the ES|QL grammars (lexer and parser) to match the latest version in Elasticsearch.'
+
+  # Check if a PR already exists
+  pr_search_result=$(gh pr list --search "$PR_TITLE" --state open --author "$KIBANA_MACHINE_USERNAME"  --limit 1 --json title -q ".[].title")
+
+  if [ "$pr_search_result" == "$PR_TITLE" ]; then
```

Comment:
If my understanding is correct here, if a PR exists it won't get updated with further changes but it requires to wait for a merge + new CI job to get new changes. Am I correct?

I do not think that's a main concern, just want to understand the flow.

Reply:
Correct. There's no logic to update the existing PR. Something we could always add in the future if this doesn't work for us!